### PR TITLE
add workspace helper

### PR DESCRIPTION
**Pull request by jrhizor**
Add the ability to pull and cache workspace information for common resources.

### CODE REVIEW

File: airbyte-server/src/main/java/io/airbyte/server/helpers/WorkspaceHelper.java [hunk 1]
```diff
@@ -0,0 +1,155 @@
...
+    return connectionToWorkspaceCache.get(connectionId);
+  }
+
+  public UUID getWorkspaceForOperationId(UUID operationId) {
+    throw new NotImplementedException();
+  }
+
+  private static <K, V> LoadingCache<K, V> getExpiringCache(CacheLoader<K, V> cacheLoader) {
+    return CacheBuilder.newBuilder()
+        .expireAfterAccess(5, TimeUnit.MINUTES)
```
Comment (cgardens) [inline airbyte-server/src/main/java/io/airbyte/server/helpers/WorkspaceHelper.java:151]: what's the merit of a temporal expiration? i think all of these cases are where these ids should never change. if there is some reason that they are changing it's going to be kinda big deal and a 5 minute expiration is not really going to help. 

my sense is we probably don't want a temporal expiration? after that we can decide if we need manual expiration or not expiration. 

what do you think? am i way off here?

**Review by cgardens — CHANGES_REQUESTED** [review]
one questions on the cache that i want to resolve. should be quick. after that it's an approve.

File: airbyte-server/src/main/java/io/airbyte/server/helpers/WorkspaceHelper.java (hunk 1 above)
Comment (jrhizor) [reply airbyte-server/src/main/java/io/airbyte/server/helpers/WorkspaceHelper.java]: I was mostly concerned about an infinitely growing cache where we don't want to keep all of the connections in memory. I switched it in https://github.com/airbytehq/airbyte/pull/4868/commits/d080a236f2da7277e148fdff2e1cece3d49d6d8e to a fixed 20k limit.

Does that sound good?

Comment (cgardens) [reply airbyte-server/src/main/java/io/airbyte/server/helpers/WorkspaceHelper.java]: i dig it. thanks!

**Review by cgardens — APPROVED** [review]

**Pull request merged**